internal/tofu/context_plan.go

```diff
@@ -664,9 +664,18 @@ func (c *Context) planWalk(config *configs.Config, prevRunState *states.State, o
 }
 
 func (c *Context) planGraph(config *configs.Config, prevRunState *states.State, opts *PlanOpts) (*Graph, walkOperation, tfdiags.Diagnostics) {
+	var diags tfdiags.Diagnostics
```

Comment:
This whole block can be simplified by not pre-creating diags/graph and returning nil, walkPlan, forgetDiags below.  That would be equivalent to this current refactoring.

Reply:
✅